tensorflow_serving/tools/docker/Dockerfile.mkl

```diff
@@ -0,0 +1,58 @@
+ARG TF_SERVING_VERSION=latest
+ARG TF_SERVING_BUILD_IMAGE=tensorflow/serving:${TF_SERVING_VERSION}-devel-mkl
+
+FROM ${TF_SERVING_BUILD_IMAGE} as build_image
+FROM ubuntu:16.04
+
+ARG TF_SERVING_VERSION_GIT_BRANCH=master
+ARG TF_SERVING_VERSION_GIT_COMMIT=head
+
+LABEL maintainer="gvasudevan@google.com"
+LABEL tensorflow_serving_github_branchtag=${TF_SERVING_VERSION_GIT_BRANCH}
+LABEL tensorflow_serving_github_commit=${TF_SERVING_VERSION_GIT_COMMIT}
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        ca-certificates \
+        && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install TF Serving pkg
+COPY --from=build_image /usr/local/bin/tensorflow_model_server /usr/bin/tensorflow_model_server
+
+# MKL installation
+COPY --from=build_image /mkl-dnn/external/mklml_lnx_*/lib/libiomp5.so /usr/local/lib
+COPY --from=build_image /mkl-dnn/external/mklml_lnx_*/lib/libmklml_gnu.so /usr/local/lib
+COPY --from=build_image /mkl-dnn/external/mklml_lnx_*/lib/libmklml_intel.so /usr/local/lib
+
+ENV LIBRARY_PATH '/usr/local/lib:$LIBRARY_PATH'
+ENV LD_LIBRARY_PATH '/usr/local/lib:$LD_LIBRARY_PATH'
+
+
+# Expose ports
+# gRPC
+EXPOSE 8500
+
+# REST
+EXPOSE 8501
+
+# Set where models should be stored in the container
+ENV MODEL_BASE_PATH=/models
+RUN mkdir -p ${MODEL_BASE_PATH}
+
+# The only required piece is the model name in order to differentiate endpoints
+ENV MODEL_NAME=model
+
+# Set inter_op or intra_op threads
+ENV TENSORFLOW_SESSION_PARALLELISM=2
```

Comment:
Why is this set to 2? Isn't this the number of cores?

Reply:
```
CORES_PER_SOCKET=`lscpu | grep "Core(s) per socket" | cut -d':' -f2 | sed "s/ //g"`
SOCKETS=`lscpu | grep "Socket(s)" | cut -d':' -f2 | sed "s/ //g"`
TOTAL_PHYSICAL_CORES=$((${CORES_PER_SOCKET} * ${SOCKETS} ))
HALF_CORES=$(( ${TOTAL_PHYSICAL_CORES} / 2))

ENV TENSORFLOW_SESSION_PARALLELISM=${HALF_CORES}
```
Above snippet will assign half of physical cores by default, if that is okay. I will update.

Reply:
Can you just do getconf _NPROCESSORS_ONLN ?  This is less error prone. I just don't like the grep and sed.

Reply:
Oh i see that is portable, but it will give you total cores=physical+logical cores. We have to divide twice to get the actual physical core counts for the half. 🤔

Reply:
Oh, what happens when there is no logical cores?  I don't think it is a good idea to divide twice. 
Also, total cores including logical cores should be fine for number of threads it can handle, so I don't know why you can't just do half of that only.

Reply:
I think your snippet seems right given all the socket combinations possible. You should definitely add a comment about how you arrived at this value :)

Reply:
yes sure.